plugin/SearchBarItem.qml

```diff
@@ -0,0 +1,241 @@
+    signal requestNextMatch()
+    signal requestCancel()
+
+    onClicked: searchField.iconized = false
+
+    states: [State {
+                 name: "extended"
+                 when: !searchField.iconized
+                 PropertyChanges {
+                     target: searchField
+                     _margin: 0.
+                 }
+             },
+             State {
+                 name: "iconized"
+                 when: searchField.iconized
+                 PropertyChanges {
+                     target: root
+                     width: iconizedWidth
+                 }
+             }]
+    transitions: Transition {
+        NumberAnimation {
+            properties: "_margin, width"
+            easing.type: Easing.InOutQuad
+            duration: 400
+        }
+    }
+    highlighted: down || searchIcon.down
+
+    TextField {
+        id: searchField
+
+        property bool iconized: true
+        property int modelCount: -1
+
+        property real _prevNextWidth: modelCount > 0 ? (Theme.itemSizeSmall + Theme.paddingMedium)*2 : 0.
+        implicitWidth: !iconized ? _editor.implicitWidth + Theme.paddingSmall
+                                   + Theme.itemSizeSmall*2  // width of two icons
+                                   + _prevNextWidth // width of prev / next icons
+                                 : Theme.itemSizeSmall
+        height: Math.max(Theme.itemSizeMedium, _editor.height + Theme.paddingMedium + Theme.paddingSmall)
+
+        property real _margin: Math.max((root.iconizedWidth - Theme.itemSizeSmall) / 2., 0.)
+        anchors {
+            verticalCenter: parent.verticalCenter
+            left: parent.left
+            right: parent.right
+            leftMargin: _margin
+            rightMargin: _margin
+        }
+
+        focusOutBehavior: FocusBehavior.ClearPageFocus
+        font {
+            pixelSize: Theme.fontSizeLarge
+            family: Theme.fontFamilyHeading
+        }
+
+        textLeftMargin: Theme.itemSizeSmall + Theme.paddingMedium + Theme.horizontalPageMargin - Theme.paddingLarge
+        textRightMargin: Theme.itemSizeSmall + Theme.paddingMedium + Theme.horizontalPageMargin - Theme.paddingLarge + _prevNextWidth
+        textTopMargin: labelVisible ? Theme.paddingSmall : (height/2 - _editor.implicitHeight/2)
+        labelVisible: modelCount > 0 && !searchField._editor.activeFocus
+        Binding on label {
+            target: searchField
+            //% "%n item(s) found"
+            value: qsTrId("sailfish-office-lb-%n-matches", modelCount)
+            when: modelCount > 0
+        }
+
+        onModelCountChanged: if (modelCount == 0) text = "" // Allow the placeholder
+        // to be visible for no results.
+        onIconizedChanged: if (!iconized) _editor.forceActiveFocus()
+
+        //: Placeholder text of SearchField
+        placeholderText: (modelCount == 0)
+            //% "No result"
+            ? qsTrId("sailfish-office-search-no-result")
```

Comment:
Search with something that's not found and refocus text field -> No result as placeholder text. Would expect "Search on document", no? Text also as "No results" on spec jpg.